### PR TITLE
keep old behaviour in RoutePath when we just have coordinates (and no…

### PR DESCRIPTION
**Pull request by jcardus**
… positions)

include name property in both cases.

### CODE REVIEW

**Comment by jcardus** [comment]
any comments?

**Comment by tananaev** [comment]
I think we should probably split `MapRoutePath` to component that handles positions and the one that handles just coordinates.

**Comment by jcardus** [comment]
can you check?

File: src/map/MapRoutePath.js [hunk 1]
```diff
@@ -1,28 +1,13 @@
 import { useTheme } from '@mui/styles';
 import { useId, useEffect } from 'react';
-import { useSelector } from 'react-redux';
 import { map } from './core/MapView';
 import { getSpeedColor } from '../common/util/colors';
 
-const MapRoutePath = ({ name, positions, coordinates }) => {
+const MapRoutePath = ({ name, positions }) => {
```
Comment (tananaev) [inline src/map/MapRoutePath.js:6]: It looks like `name` is never used. We can probably remove it from here.

File: src/map/MapRouteCoordinates.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      layout: {
+        'line-join': 'round',
+        'line-cap': 'round',
+      },
+      paint: {
+        'line-color': ['get', 'color'],
+        'line-width': 2,
+      },
+    });
+    if (name) {
```
Comment (tananaev) [inline src/map/MapRouteCoordinates.js:46]: For this one the name is always provided, so we can probably make it mandatory.

File: src/map/MapRoutePath.js [hunk 3]
```diff
@@ -1,28 +1,13 @@
 import { useTheme } from '@mui/styles';
 import { useId, useEffect } from 'react';
-import { useSelector } from 'react-redux';
 import { map } from './core/MapView';
```
Comment (tananaev) [inline src/map/MapRoutePath.js:3]: Can you please check the warnings and errors.

Comment (jcardus) [reply src/map/MapRoutePath.js:3]: I don't have any warnings

File: src/map/MapRoutePath.js [hunk 4]
```diff
@@ -1,27 +1,13 @@
...
+import { getSpeedColor } from '../common/util/colors';
 
-const MapRoutePath = ({ name, positions, coordinates }) => {
+const MapRoutePath = ({ positions }) => {
   const id = useId();
 
   const theme = useTheme();
 
-  const reportColor = useSelector((state) => {
```
Comment (tananaev) [inline src/map/MapRoutePath.js:11]: Why are we removing this?

Comment (jcardus) [reply src/map/MapRoutePath.js:11]: the color is based on the position speed

Comment (tananaev) [reply src/map/MapRoutePath.js:11]: Yeah, but it means you're ignoring the settings.

**Pull request merged**